listenbrainz/webserver/views/feedback_api.py

```diff
@@ -0,0 +1,180 @@
+    :type score: ``int``
+    :param count: Optional, number of feedback items to return, Default: :data:`~webserver.views.api.DEFAULT_ITEMS_PER_GET`
+        Max: :data:`~webserver.views.api.MAX_ITEMS_PER_GET`.
+    :type count: ``int``
+    :param offset: Optional, number of feedback items to skip from the beginning, for pagination.
+        Ex. An offset of 5 means the top 5 feedback will be skipped, defaults to 0.
+    :type offset: ``int``
+    :statuscode 200: Yay, you have data!
+    :resheader Content-Type: *application/json*
+    """
+
+    score = _parse_int_arg('score')
+
+    offset = _get_non_negative_param('offset', default=0)
+    count = _get_non_negative_param('count', default=DEFAULT_ITEMS_PER_GET)
+
+    count = min(count, MAX_ITEMS_PER_GET)
+
+    user = db_user.get_by_mb_id(user_name)
+    if user is None:
+        raise APINotFound("Cannot find user: %s" % user_name)
+
+    args = {}
+    args["user_id"] = user["id"]
+    args["limit"] = count
+    args["offset"] = offset
+
+    if score:
+        if score not in [-1, 1]:
+            log_raise_400("Score can have a value of 1 or -1.", request.args)
+        args["score"] = score
+
+    feedback = db_feedback.get_feedback_for_user(**args)
+    total_count = db_feedback.get_feedback_count_for_user(user["id"])
+
+    for i, fb in enumerate(feedback):
+        fb.user_id = fb.user_name
+        del fb.user_name
+        feedback[i] = fb.dict()
+
+    return jsonify({
+        "feedback": feedback,
+        "count": len(feedback),
+        "total_count": total_count,
+        "offset": offset
+    })
+
+
+@feedback_api_bp.route("/recording/<recording_msid>/get-feedback", methods=["GET"])
+@crossdomain()
+@ratelimit()
+def get_feedback_for_recording(recording_msid):
+    """
+    Get feedback for recording with given ``recording_msid``. The format for the JSON returned
+    is defined in our :ref:`feedback-json-doc`.
+
+    :param score: Optional, If 1 then returns the loved recordings, if -1 returns hated recordings.
+    :type score: ``int``
+    :param count: Optional, number of feedback items to return, Default: :data:`~webserver.views.api.DEFAULT_ITEMS_PER_GET`
+        Max: :data:`~webserver.views.api.MAX_ITEMS_PER_GET`.
+    :type count: ``int``
+    :param offset: Optional, number of feedback items to skip from the beginning, for pagination.
+        Ex. An offset of 5 means the top 5 feedback will be skipped, defaults to 0.
+    :type offset: ``int``
+    :statuscode 200: Yay, you have data!
+    :resheader Content-Type: *application/json*
+    """
+
+    if not is_valid_uuid(recording_msid):
+        log_raise_400("%s MSID format invalid." % recording_msid)
+
+    score = _parse_int_arg('score')
+
+    offset = _get_non_negative_param('offset', default=0)
+    count = _get_non_negative_param('count', default=DEFAULT_ITEMS_PER_GET)
+
+    count = min(count, MAX_ITEMS_PER_GET)
+
+    args = {}
```

Comment:
This seems hard to understand to me. Could we keep the args as variables and pass them explicitly into the function? That would be much more readable.

Reply:
Same for the other endpoint

Reply:
Fixed